devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -0,0 +1,572 @@
+		command "increaseHeatSetpoint"
+		command "decreaseHeatSetpoint"
+		command "setCustomThermostatMode"
+		command "eco"
+		command "updateWeather"
+
+		fingerprint deviceId: "0x0806", inClusters: "0x5E,0x86,0x72,0x40,0x43,0x31,0x85,0x59,0x5A,0x73,0x20,0x42", mfr: "0239", prod: "0001", model: "0001", deviceJoinName: "Stelpro Ki Thermostat"
+	}
+
+	// simulator metadata
+	simulator { }
+
+	preferences {
+		input("heatdetails", "enum", title: "Do you want a detailed operating state notification?", options: ["No", "Yes"], defaultValue: "No", required: true, displayDuringSetup: true)
+		input("zipcode", "text", title: "ZipCode (Outdoor Temperature)", description: "[Do not use space](Blank = No Forecast)")
+	}
+
+	tiles(scale : 2) {
+		multiAttributeTile(name:"thermostatMulti", type:"thermostat", width:6, height:4, canChangeIcon: true) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState("temperature", label:'${currentValue}°')
+			}
+			tileAttribute("device.heatingSetpoint", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "increaseHeatSetpoint")
+				attributeState("VALUE_DOWN", action: "decreaseHeatSetpoint")
+			}
+			tileAttribute("device.thermostatOperatingState", key: "OPERATING_STATE") {
+				attributeState("idle", backgroundColor:"#44b621")
+				attributeState("heating", backgroundColor:"#ffa81e")
+			}
+			tileAttribute("device.thermostatMode", key: "THERMOSTAT_MODE") {
+				attributeState("heat", label:'${name}')
+				attributeState("eco", label:'${name}')
+			}
+			tileAttribute("device.heatingSetpoint", key: "HEATING_SETPOINT") {
+				attributeState("heatingSetpoint", label:'${currentValue}°')
+			}
+		}
+		standardTile("mode", "device.thermostatMode", width: 2, height: 2) {
+			state "heat", label:'${name}', action:"switchMode", nextState:"eco", icon:"http://cdn.device-icons.smartthings.com/Home/home29-icn@2x.png"
+			state "eco", label:'${name}', action:"switchMode", nextState:"heat", icon:"http://cdn.device-icons.smartthings.com/Outdoor/outdoor3-icn@2x.png"
+		}
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2) {
+			state "heatingSetpoint", label:'Setpoint ${currentValue}°', backgroundColors:[
+					[value: 31, color: "#153591"],
+					[value: 44, color: "#1e9cbb"],
+					[value: 59, color: "#90d2a7"],
+					[value: 74, color: "#44b621"],
+					[value: 84, color: "#f1d801"],
+					[value: 95, color: "#d04e00"],
+					[value: 96, color: "#bc2323"]
+			]
+		}
+		standardTile("refresh", "device.refresh", decoration: "flat", width: 2, height: 2) {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		main ("thermostatMulti")
+		details(["thermostatMulti", "mode", "heatingSetpoint", "refresh"])
+	}
+}
+
+def getSupportedThermostatModes() {
+	modes()
+}
+
+def getThermostatSetpointRange() {
+	if (getTemperatureScale() == "C") {
+		[5, 30]
+	}
+	else {
+		[41, 86]
+	}
+}
+
+def getHeatingSetpointRange() {
+	thermostatSetpointRange
```

Comment:
:/

are these functions needed somewhere else that they're just wrappers for identical functions?

Reply:
I suppose since this is a heater only I could just use the one function. modes() existed before in the code as a helper function, but I created these (well, copied these from another thermostat DTH) to try to be at least somewhat consistent.